community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetector.java

```diff
@@ -0,0 +1,139 @@
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/**
+ * A {@link VulnDetector} that detects the CVE-2020-14882 vulnerability.
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "Cve202014882VulnDetector",
+    version = "0.1",
+    description =
+        "Vulnerability in the Oracle WebLogic Server product of Oracle Fusion Middleware (component: Console). "
+            + "Supported versions that are affected are 10.3.6.0.0, 12.1.3.0.0, 12.2.1.3.0, 12.2.1.4.0 and 14.1.1.0.0. "
+            + "Easily exploitable vulnerability allows unauthenticated attacker with network access via HTTP to compromise "
+            + "Oracle WebLogic Server. Successful attacks of this vulnerability can result in takeover of Oracle WebLogic Server",
+    author = "thiscodecc",
+    bootstrapModule = Cve202014882DetectorBootstrapModule.class)
+public class Cve202014882VulnDetector implements VulnDetector {
+
+  @VisibleForTesting
+  static final String DETECTION_STRING = "/console/jsp/common/warnuserlockheld.jsp";
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String CHECK_VUL_PATH = "console/css/%252e%252e%252fconsole.portal?_nfpb=true&_pageLabel=HomePage1";
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+
+  @Inject
+  Cve202014882VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient).modify().setFollowRedirects(false).build();
+    this.utcClock = checkNotNull(utcClock);
+  }
+
+  private static boolean isWebServiceOrUnknownService(NetworkService networkService) {
+    return networkService.getServiceName().isEmpty()
+        || NetworkServiceUtils.isWebService(networkService)
+        || NetworkServiceUtils.getServiceName(networkService).equals("unknown");
+  }
+
+  @Override
+  public DetectionReportList detect(TargetInfo targetInfo,
+      ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Cve202014882VulnDetector starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream().filter(Cve202014882VulnDetector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + CHECK_VUL_PATH;
+
+    try {
+      HttpResponse httpResponse =
+          httpClient.send(get(targetUri).withEmptyHeaders().build(), networkService);
+
+      //Solve the WebLogic http 302 redirection problem
+      Optional<String> cookie = httpResponse.headers().get("Set-Cookie");
```

Comment:
Please use `com.google.common.net.HttpHeaders.SET_COOKIE`, similar to the `COOKIE` header on line 97.